src/harness/testcases/WINNF_FT_S_IPR_testcase.py

```diff
@@ -0,0 +1,244 @@
+
+    # Trigger CPAS.
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Register N3 CBSDs with the SAS UUT and request Grants.
+    n3_domain_proxy = DomainProxy(self)
+    n3_domain_proxy.registerCbsdsAndRequestGrants(
+        config['registrationRequestsN3'],
+        config['grantRequestsN3'],
+        conditional_registration_data=config['conditionalRegistrationDataN3'])
+
+    # Heartbeat all SAS UUT CBSDS
+    n2_domain_proxy.heartbeatForAllActiveGrants()
+    n3_domain_proxy.heartbeatForAllActiveGrants()
+
+    # None of the N3 CBSDS should be authorized.
+    for cbsd in n3_domain_proxy.getCbsdsWithAtLeastOneAuthorizedGrant():
+      for operation_param in cbsd.getOperationParamsOfAllAuthorizedGrants():
+        # By definition the N3 CBSDs are in a DPA neighborhood. Check frequency.
+        self.assertFalse(
+            self.frequencyInBand(
+                operation_param['operationFrequencyRange']['lowFrequency'],
+                operation_param['operationFrequencyRange']['highFrequency']),
+            msg='CBSD (cbsd_id=%s, fcc_id=%s, sn=%s) '
+            'is authorized after IPR.1 step 7. SAS UUT FAILS this test. '
+            '(If this config is new please verify the CBSD is in a DPA)' %
+            (cbsd.getCbsdId(), cbsd.getRegistrationRequest()['fccId'],
+             cbsd.getRegistrationRequest()['cbsdSerialNumber']))
+
+    # Get SAS UUT FAD and Test Harness FADs.
+    sas_uut_fad = getFullActivityDumpSasUut(self._sas, self._sas_admin)
+    test_harness_fads = []
+    for test_harness in test_harnesses:
+      test_harness_fads.append(
+          getFullActivityDumpSasTestHarness(
+              test_harness.getSasTestHarnessInterface()))
+
+    # Trigger CPAS.
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Initialize DPA objects and calculate movelist for each DPA.
+    dpas = []
+    for dpa_config in config['dpas']:
+      dpa = dpa_mgr.BuildDpa(dpa_config['dpaId'], dpa_config['points_builder'])
+      freq_range_low = dpa_config['frequencyRange']['lowFrequency'] / ONE_MHZ
+      freq_range_high = dpa_config['frequencyRange']['highFrequency'] / ONE_MHZ
+      dpa.ResetFreqRange([(freq_range_low, freq_range_high)])
+      dpa.SetGrantsFromFad(sas_uut_fad, test_harness_fads)
+      dpa.ComputeMoveLists()
+      dpas.append(dpa)
+    # Heartbeat SAS UUT grants.
+    n2_domain_proxy.heartbeatForAllActiveGrants()
+    n3_domain_proxy.heartbeatForAllActiveGrants()
+    # Get CbsdGrantInfo list of SAS UUT grants that are in an authorized state.
+    # TODO: Update with correct function name/import when it is added.
+    grant_info = getGrantsFromDomainProxies([n2_domain_proxy, n3_domain_proxy])
+
+    # Check grants do not exceed each DPAs interference threshold.
+    for dpa, dpa_config in zip(dpas, config['dpas']):
+      freq_range_low = dpa_config['frequencyRange']['lowFrequency'] / ONE_MHZ
```

Comment:
You can now change this into:
```
for dpa in dpas:
   self.assertTrue(  dpa.checkInterference(sas_uut_active_grants=grant_info,  margin_db=margin_db)
``` 
Note that the signature has changed and the channel is on 3rd position if you want to continue using it.
Please note also that the margin_db should not be per DPA, I believe, but rather a general parameter. Although I see why you want to make it configurable in the config test, so probably ok your way.

Reply:
Added change to IPR.2 PR.